src/validation.cpp

```diff
@@ -4369,7 +4369,7 @@ std::string GetWarnings(const std::string& strFor)
 
 ThresholdState VersionBitsTipState(const Consensus::Params& params, Consensus::DeploymentPos pos)
 {
-    LOCK(cs_main);
+    AssertLockHeld(cs_main);
```

Comment:
While I agree that `AssertLockHeld` should be enough here, it feels rather unrelated to DIP0001 specifically, so I think its better to keep it as is for now. Revert this pls.

Reply:
fixed (moved this change to a separate PR)